Use almstars catalogues to improve Almagest sky culture

abalkin opened the pull request:
## Catalogues

1. Toomer/Grasshoff  (cat1.dat)
Data for this catalog was transcribed from Appendix B of Grasshoff's The
History of Ptolemy's Star Catalog (Springer-Verlag, 1990). With minor
variations, this is based on Toomer's translation of the Almagest, first
edition (Duckworth, 1984). The transcriptions were reconciled with the
second edition of Toomer (Princeton University Press, 1998), which is
also the source of the star descriptions.

2. Peters/Knobel  (cat2.dat)
Transcribed from Peters and Knobel, Ptolemy's Catalogue of Stars
(Carnegie Institution, 1915).

3. Manitius  (cat3.dat)
Initially CDS V/61, the electronic catalog by Jaschek, but substantially
corrected by consulting the source, Manitius's 1913 German translation
of the Almagest, the (Teubner, 1962) edition with the forward by
Neugebauer. The star descriptions, not present in V/61, were transcribed
directly from Manitius.

4. Pickering (catpick.dat)

See http://www.etwright.org/astro/almagest.html#cat

<!--- Provide a general summary of your changes in the Title above -->

### Description
<!--- Please include a summary of the change and which issue is fixed. -->
<!--- Please also include relevant motivation and context. -->
<!--- List any dependencies that are required for this change. -->

Fixes #1646

### Screenshots (if appropriate):

### Type of change
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] This change requires a documentation update

### How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

**Test Configuration**:
* Operating system: <Name, version number>
* Graphics Card: <Manufacturer (likely Intel, NVidia, AMD?), Model (HD, Geforce, Radeon..., with model number), driver version?>

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the [code style](http://stellarium.org/doc/head/codingStyle.html) of this project.
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

gzotti commented on skycultures/almagest/make_names.py:
```diff
@@ -0,0 +1,51 @@
+            for tag in tags:
+                print(f'{tag}|_("{desc}")', file=dso)
+        else:
+            print(f'No HIP number found for HR {hr} ({modern_id})', file=sys.stderr)
```
please add newline at EOF

abalkin replied:
done

gzotti reviewed (changes requested):
The PR is flagged as "needs change in documentation". What exactly has changed in the documentation?

I see additions of source catalogs (are they OK to distribute via FOSS or do they have their own licenses?) and some Python tools that probably extract the star names from the respective catalogs. What does the user do with this? Are the tools geared towards the ordinary user (who will have to run them as superuser/Administrator as the skycultures goes in protected install directories; this must be documented in the Skycultures chapter of the User Guide) or are the tools for the developers of this SC, and the result is delivered to the end users (This must be documented in a README.md here)?

I don't talk bad about this effort, I think it's valuable work, but it must become clearer what we are adding here and why. As said before, this may also be seen as starting point for the "user star list" plugin, providing some catalogs and sample code.

alex-w commented:
> The PR is flagged as "needs change in documentation". What exactly has changed in the documentation?

This flag was put by me, because proposed changes at least require updating description of SC and list of sources. But okay, I’ve removed this flag now.

abalkin commented:
I switched back to draft because we should be consistent with the Al-Sufi skylore and turn long descriptions into "additional" rather than primary names.

abalkin commented:
@gzotti - did I address all your comments?

sushoff commented:
> I switched back to draft because we should be consistent with the Al-Sufi skylore and turn long descriptions into "additional" rather than primary names.

Don't think so! 1. We should not do it consistent with XY because we want to show historical development. 2. I want the primary name to be the original one.

abalkin commented:
> 1. We should not do it consistent with XY because we want to show historical development. 

@sushoff  - al-Sufi in the context of Almagest is not just "XY", it is a very closely related culture.  In fact, @kajaji, the author of the Al-Sufi culture implementation used the same Ernie Wright's files to obtain English translations as I do here. "The translation of star descriptions from Arabic into English was extracted from Ernie Wright website[6] which was transcribed from Toomer's translation of the Almagest[7]."  (See [skycultures/al-sufi/description.en.utf8](https://github.com/Stellarium/stellarium/blob/fb25a4bae5d9616bdbcb43cd16e4e8000bd10b19/skycultures/al-sufi/description.en.utf8#L63)). The "Con n" (or "Con_n") abbreviations do not come from historical sources.  Ernie Wright added IAU-style abbreviations to his tables in order to preserve the groupings of stars by constellations.  The ordinal numbers within constellations are also not present in Almagest and the Book of Fixed Stars uses abjad numerals.   

> 2. I want the primary name to be the original one.

With very few exceptions, there is no such thing as a *name* of a star in Almagest.  I was able to find 11 proper names in Toomer:  Arcturus (α Boo), Lyra (α Lyra), Capella (α Aur),  Aquila (α Aql), Praesepe (M 44), Regulus (α Leo),  Spica (α Vir), Vindemiatrix (ε Vir),  Antares (α Sco), "the Dog" (Sirius, α CMa) and Procyon (α CMi).  Ptolemy's descriptions are not suitable for primary names for at least two reasons: (1) they are too long (e.g. "The most advanced of the 3 stars on the spine in the back, following [i.e. to the rear of] the star on the elbow of Andromeda") and (2) they often refer to a previously defined group (e.g. there are 18 stars described as "The middle one of these").  Note that I don't have a solution for the second issue yet, but the most likely solution is to repeat the definition of the group in each description. For example, ꭓ Oph can be described as "The middle one of  [the 3 stars in a straight line in the left lower leg]".  This, however will lengthen the already long descriptions make the first issue worse.

Overall, I find the use of invented abbreviations to be the only practical solution if we want to provide complete Almagest-like sky cultures.  Think of "Con n" labels as a kind of footnote marker inviting the user to click to see a longer description.  I'll be happy to entertain alternatives as long as they are reasonably short and can be systematically applied to all stars in the catalogue.

sushoff commented:
> Ptolemy's descriptions are not suitable for primary names for at least two reasons: (1) they are too long

yes, I agree

but in as-Sufi they are used anyway: I am still translating this catalogue for Stellarium

abalkin commented:
> but in as-Sufi they are used anyway: I am still translating this catalogue for Stellarium

@sushoff - which version of Stellarium are you looking at?  With the latest master, I see the following when al-Sufi skylore is selected:

![image](https://user-images.githubusercontent.com/535197/118020129-bebbb080-b327-11eb-8a81-fe339c2c0d62.png)

As you can see, the display is similar to what I implemented for Almagest.  (See [my note](https://github.com/Stellarium/stellarium/issues/1646#issuecomment-839777462) in #1646. )

English translations of secondary names do seem to be incomplete, but they are not used as star labels – only as additional names.

abalkin commented:
>  I am still translating this catalogue for Stellarium

@sushoff - have you seen [Hafez, Ihsan (2010) translation](https://researchonline.jcu.edu.au/28854/) of sections of Al-Ṣūfī's 'Book of the Fixed Stars'?  It appears that the author placed that work in the public domain and there should be no issues with copying the names of the stars from there.  To give you a flavor, here is a discussion of the seven UMi stars:

> As for the Small (Bear), the Arabs call the seven (stars) of the group Banāt Na’sh al-Ṣugrha (The Little Daughters of the Bier or Coffin). The four on the rectangle are the Na’sh (bier or coffin) and the three on the tail are the Banāt (the daughters). (The Arabs) call the two brightest stars of the rectangle al- Farqadain and the bright star on the end of the tail al-Juday; this (star) is used to locate the Kiblah (= the direction of Makkah).
The three stars on the tail, together with the fourth and sixth (the word “seventh” has been scratched out) form a curved line. Next to the brightest (star) of the al-Farqadain, which is the sixth, is a fainter star, on the same line as the al-Farqadain (but) not of the constellation.  --  Hafez, Ihsan (2010), page 112.

sushoff commented:
sorry, I mean, I am translating the (new) SC from English to German (sorry, did not make for the last release)

kajaji commented:
> > I am still translating this catalogue for Stellarium
> 
> @sushoff - have you seen [Hafez, Ihsan (2010) translation](https://researchonline.jcu.edu.au/28854/) of sections of Al-Ṣūfī's 'Book of the Fixed Stars'? It appears that the author placed that work in the public domain and there should be no issues with copying the names of the stars from there. To give you a flavor, here is a discussion of the seven UMi stars:
> 
> > As for the Small (Bear), the Arabs call the seven (stars) of the group Banāt Na’sh al-Ṣugrha (The Little Daughters of the Bier or Coffin). The four on the rectangle are the Na’sh (bier or coffin) and the three on the tail are the Banāt (the daughters). (The Arabs) call the two brightest stars of the rectangle al- Farqadain and the bright star on the end of the tail al-Juday; this (star) is used to locate the Kiblah (= the direction of Makkah).
> > The three stars on the tail, together with the fourth and sixth (the word “seventh” has been scratched out) form a curved line. Next to the brightest (star) of the al-Farqadain, which is the sixth, is a fainter star, on the same line as the al-Farqadain (but) not of the constellation.  --  Hafez, Ihsan (2010), page 112.

@abalkin @sushoff The [Arabic] names of the stars are already present in Al-Sufi SC, either as another secondary name after the long description or as an asterism name if they are a group of stars.
![Sec_Name](https://user-images.githubusercontent.com/34771982/118052335-366ef700-b38b-11eb-98c4-10363182cd87.JPG)

axd1967 commented:
> I see additions of source catalogs (are they OK to distribute via FOSS or do they have their own licenses?) and some Python tools that probably extract the star names from the respective catalogs. What does the user do with this? Are the tools geared towards the ordinary user (who will have to run them as superuser/Administrator as the skycultures goes in protected install directories; this must be documented in the Skycultures chapter of the User Guide) or are the tools for the developers of this SC, and the result is delivered to the end users (This must be documented in a README.md here)?

If external code/text/data is included, please try to implement the Vendor Branch mechanism rather than just copy-pasting external code into the repository. The VB concept is simple and guarantees that external future code updates can easily be imported. Please do this to make maintenance of such code easier. (Remember: _copy-paste is evil_.)

In this case, the external data (and everything that is copied from outside Stellarium) should be vendor branched from

http://www.etwright.org/astro/almagest.html#catfmt

Note that the README and/or AUTHORS **should** contain that URL to avoid people having to search for it, and to monitor for possible changes. Even if the source data has not changed since 2007!

For details, see **the last answer** of  https://stackoverflow.com/questions/769786/vendor-branches-in-git . (Avoid git submodules.)

abalkin commented:
I added proper names that can be found in Toomer's translation.  When available, these will be displayed instead of synthetic 'Con n' tags.

![image](https://user-images.githubusercontent.com/535197/118381211-972e4780-b5b6-11eb-80fc-f82a08d62538.png)

Reviewers (@sushoff,  @gzotti , and @alex-w), please indicate your approval or tell me what changes need to be made before this PR can be accepted.

kajaji commented:
> I added proper names that can be found in Toomer's translation. When available, these will be displayed instead of synthetic 'Con n' tags.
> 
> ![image](https://user-images.githubusercontent.com/535197/118381211-972e4780-b5b6-11eb-80fc-f82a08d62538.png)
> 
> Reviewers (@sushoff, @gzotti , and @alex-w), please indicate your approval or tell me what changes need to be made before this PR can be accepted.

I would prefer to use tags as primary names all along. Working with Al-Sufi book and Almagest, I found that It is helpful for the researcher to know the star index number in the constellation. Moreover, I think it is useful to indicate if the star is "amorphotoi" being not part of the constellation figure.

abalkin commented:
@kajaji — I don't really have a personal preference, to I will look for you and @sushoff to come to a consensus and advise.  I would just note that star indices within constellations are "native" to al-Sufi but in Almagest they only appear in late translations.  (Likely first introduced by Manitius.)

alex-w commented:
@abalkin @kajaji I've added support untranslatable designations for stars and DSOs in skycultures, because this is really need for Al-Sufi and Almagest skycultures. 

For example, line `80463|_("Her 30")` is translatable, and line `80463|("Her 30")` is not translatable

sushoff commented:
> I would like to update the culture description to give proper credit to Ernie Write, but this can be done in a separate PR.

yes please do so; I appreciate your work because I wanted to improve this SC since >1 year but did not have the time

sushoff commented:
> . The current behavior when Almagest is selected as the starlore is to display the descriptions of very few stars.

the current SC displays no descriptions at all but only single star names (such as "Arkturos"); we wanted to leave the star names (i.e. the long designators that are describing the position in the figure) for later because they are so unhandily long. The example labels that you are displaying in the star chart above are not real star names but these are names of groups "outside the constellation". It might appear as an only technical difference but I wanted to explain the concept. 

As you found a solution for the long star names, your addition is most welcome.

sushoff commented:
> > 1. We should not do it consistent with XY because we want to show historical development.
> 
> @sushoff - al-Sufi in the context of Almagest is not just "XY", it is a very closely related culture. In fact, @kajaji, the author of the Al-Sufi culture implementation used the same Ernie Wright's files to obtain English translations as I do here. "The translation of star descriptions from Arabic into English was extracted from Ernie Wright website[6] which was transcribed from Toomer's translation of the Almagest[7]." (See [skycultures/al-sufi/description.en.utf8](https://github.com/Stellarium/blob/fb25a4bae5d9616bdbcb43cd16e4e8000bd10b19/skycultures/al-sufi/description.en.utf8#L63)). 

I know the dependencies and the history of star catalogues! (have a doctoral degree in this topic). I was just killing the argument by trying to say that there is no need to make one Stellarium SC consistent with another one. Of course, it is not forbidden to copy a well-working / practical mode - but there is simply no need for consistency in various SCs. 

> The "Con n" (or "Con_n") abbreviations do not come from historical sources.

Exactly, that's what I am referring to: In the Almagest tradition something like this is NOT historical but in the Far Eastern tradition it IS historical: Chinese star "names" are in fact numbers. I am not sure whether or not this is relevant for applying your suggested counting of stars according to Toomer as primary description. On the one hand, nobody of us (I guess, you too) wants to imply that Almagest star names had numbers, on the other hand, people could assume this when they simply see the displayed label "UMi 1". However, "UMi" is modern (IAU-made) abbreviation and we could write in the description what we mean. Still, I would not automatically expect the users to think about this while using the software and even less I would expect them to read the description. Therefore, I was expressed scepticism and suggested to think about the style of the abbreviated label. 

However, as I am thinking about this since a while already, I suggest to simply take this way of labelling (Con n) because it will probably never be possible to find a completely bijective and unmisunderstandable way of communication. 

>Ptolemy's descriptions [...] (2) they often refer to a previously defined group (e.g. there are 18 stars described as "The middle one of these"). Note that I don't have a solution for the second issue yet, but the most likely solution is to repeat the definition of the group in each description. 

Exactly, this is what Toomer, Grasshoff and myself did in catalogue publications (tables) but I agree, it is not practical for display

sushoff commented:
Yet another comment/ idea: In my own working for research, I display the unhandy names as tooltips. 

This might be also a solution for Stellarium but would require additional programming solutions.

alex-w commented:
> Yet another comment/ idea: In my own working for research, I display the unhandy names as tooltips.
> 
> This might be also a solution for Stellarium but would require additional programming solutions.

This is very, very difficult to implement.

abalkin commented:
> @abalkin @kajaji I've added support untranslatable designations for stars and DSOs in skycultures, because this is really need for Al-Sufi and Almagest skycultures.
> 
> For example, line `80463|_("Her 30")` is translatable, and line `80463|("Her 30")` is not translatable

I remove underscores from "Con n" entries.   See 85e7351e202c6abce4dd5807288e8f87b66da719 .

abalkin commented:
> > I would like to update the culture description to give proper credit to Ernie Write, but this can be done in a separate PR.
> 
> yes please do so;

I added a brief mention of Ernie Write's work in the description and added a link to his web page in the list of references. We may at some point need to rewrite the "How this sky culture was made" section, but for now, I prefer to keep developers' notes in the README.md file.

sushoff commented:
> > > I would like to update the culture description to give proper credit to Ernie Write, but this can be done in a separate PR.
> > 
> > 
> > yes please do so;
> 
> I added a brief mention of Ernie Write's work in the description and added a link to his web page in the list of references. We may at some point need to rewrite the "How this sky culture was made" section, but for now, I prefer to keep developers' notes in the README.md file.

please feel free to add your contribution there (in the description); it does not change what the students made, it just adds. 

pleaes, then, also add your name in the  authors section - thank you

gzotti commented on skycultures/almagest/README.md:
```diff
@@ -0,0 +1,55 @@
+
+### The make_names.py script
+
+The `make_names.py` takes the `cat1.dat` file and generates `star_names.fab` and `dso_names.fab` files that can be used by Stellarium.  Since `star_names.fab` requires HIP rather than HR numbers, the required cross-catalogue data is extracted from the `cross-id.dat` file.  The DSOs are identified by their modern designations and the necesary translations are hard-coded in the script. (There are o ...
```
Typo: necessary

gzotti reviewed (commented):
For me it still is not clear why the 4 versions of the catalog plus an extractor program that processes cat1 only (hardcoded) are here. Don't get me wrong, catalogs and extractor are interesting. But who is supposed to run this command ever again, and what do I have to do as advanced user who is building from source if e.g. I want Manitius' version shown? Would it be easy to change the Python program to take the catalog as argument? (Are the catalogs completely compatible this way?)

kajaji commented:
> > @abalkin @kajaji I've added support untranslatable designations for stars and DSOs in skycultures, because this is really need for Al-Sufi and Almagest skycultures.
> > For example, line `80463|_("Her 30")` is translatable, and line `80463|("Her 30")` is not translatable
> 
> I removed underscores from "Con n" entries. See [85e7351](https://github.com/Stellarium/stellarium/commit/85e7351e202c6abce4dd5807288e8f87b66da719) .

@alex-w IMHO these entries should be translatable for the sake of non-western alphabet languages. For example a casual Arab user will not relate "Her 30" to the corresponding Arabic name "Al-Jathi".

To minimise the translators effort we can use the same designation "Con nn" rather than "Con_nn" as used in Al-Sufi in both sky cultures.

abalkin commented:
> @alex-w IMHO these entries should be translatable for the sake of non-western alphabet languages. For example a casual Arab user will not relate "Her 30" to the corresponding Arabic name "Al-Jathi".

@kajaji - I think you are looking at this trough the prism of al-Sufi rather than Almagest SC.  In al-Sufi SC, translating "Con n" labels to Arabic would make perfect sense because you can us Abjad numerals for the 'n' part which are native to the source.  In Almagest SC, these labels are purely navigational and don't have any historical significance. 

What I can do is include the list of abbreviations together with full constellation names in the description.  If we do it in say 4-column table, the result will be relatively short (11 rows).  This way, once description is translated to users' language, they will have a handy reference to look up.

abalkin commented:
> For me it still is not clear why the 4 versions of the catalog plus an extractor program that processes cat1 only (hardcoded) are here.

@gzotti - you are right, at the moment there is no reason to keep sources other than `cat1.dat`.  I was originally planning to use `cat3.dat` to generate German translations, but then realized that cross-reference between cat1 and cat3 is not straightforward and before `cat3.dat` is used that way someone who fluent is German should confirm that the order of descriptions is the same in the English and German catalogues.

On the other hand, the included catalogue files are so small that I don't see any downside in committing all of them for future use.  Note that to get to the source one has to download  Ernie Write's application and extract the data files from it.  They are much more accessible when included here.

> But who is supposed to run this command ever again?

Whoever wants to make changes to `dso_names.fab` and `star_names.fab`.   Should I put a comment at the top of each of these files directing the future developers to the `make_names.py` script and advising them not to change them manually?

> and what do I have to do as advanced user who is building from source if e.g. I want Manitius' version shown?

As I mentioned before, in the future we can incorporate running `make_names.py` in the build process and delete the generated.fab files from the source distribution.   If this is done, we can make a choice of source catalogue a  build option. At the moment an advanced user can change the `CAT_FILE` constant in the script.  (Would you like me to bring it to the top of the file?)

>  Would it be easy to change the Python program to take the catalog as argument?

This is very easy, but I am not sure this will be an improvement.

>  (Are the catalogs completely compatible this way?)

Yes, all four catalogues have the same format, but I did not test my script with any of the alternative catalogue files yet.

sushoff commented:
> > For me it still is not clear why the 4 versions of the catalog plus an extractor program that processes cat1 only (hardcoded) are here.
> 
> > But who is supposed to run this command ever again?
> 
> Whoever wants to make changes to `dso_names.fab` and `star_names.fab`. Should I put a comment at the top of each of these files directing the future developers to the `make_names.py` script and advising them not to change them manually?
> 


I find this very helpful and I would like to publish it elsewhere, where it reaches the right community who will use it. (one of the conference proceedings from either last year or this year).  The newly developed format of "computational books" (cBooks) shall be used for this.

alex-w commented:
> > But who is supposed to run this command ever again?
> 
> Whoever wants to make changes to `dso_names.fab` and `star_names.fab`. Should I put a comment at the top of each of these files directing the future developers to the `make_names.py` script and advising them not to change them manually?

Please add instructions and notifications into README.md

sushoff commented:
I don't like your change of the name of M44. You just copied Toomer's translation which is not the original Greek: Toomer translates "called Praesepe" but Ptolemy wrote in Greek, which is why he did not call the object Praesepe (latin): That is why, I used the original Greek term! That is why I insist to use my version ("φάτνη (Manger)") (Greek-English) instead of using the now common Latin word (which was ok in Toomer's book but not here). If you want to add the whole description, that's fine but don't translate Greek with Latin! I suggest:  

("The middle of the nebulous mass in the chest, called φάτνη (Manger)")

alex-w commented:
> @alex-w IMHO these entries should be translatable for the sake of non-western alphabet languages. For example a casual Arab user will not relate "Her 30" to the corresponding Arabic name "Al-Jathi".

Right now the "Her 30" string is just designation, which shouldn't be translated IMHO. Should be name for these stars as "Constellation nn"?

> To minimise the translators effort we can use the same designation "Con nn" rather than "Con_nn" as used in Al-Sufi in both sky cultures.

Maybe. I've added the ability to mark some names as untranslatable names (I've mark "Con_nn" as untranslatable for al-Sufi SC, sorry), but you may do not use them - this is optional feature.

@gzotti @sushoff what about sync the "designations" for 2 SC?

gzotti commented:
I welcome the "Con nn" as navigational aid when I have the Almagest on my desk in front of the screen, and I cannot say if they (just the IAU designations) even need transliteration. (Yet another 1028 strings?) Are all known editions alike in this respect? Then it would make sense to have the same labels in Almagest and Sufi SCs. @kajaji These are not names in any classical edition, but help modern users. These short labels may also be better as primary star labels shown in the sky. Selecting one star then shows the potentially very long description in the Infostring. 

Suggestion: Maybe mark the "unformed" stars separately, like "Boo U1"? (Like the Chinese "added" designation.) Or as "Boo 23U" This could make it clearer at once if the star is part of the actual figure. But this information may not be in the almstar lists.

Yes, what to do with the other catalogs and how to actually run the Python program (or that you have to edit it for the other catalogs) should go into the README.md. Expect users to know nothing about Python.

As noted earlier, the catalogs in this form may become parts of a future "UserStarCatalog" plugin for scientific users and may be moved there at some later date. I did not know this almstars program, but it looks quite a lot of what I had in mind. 

Are the catalogs strict GPL2 only? Maybe Mr. Wright can release them (data items) under a CC license as well? It may make things easier in the future. Also, German "Umlaut" characters are missing in the Manitius list (cat3), this should be changed at some later date.

The replacement of @sushoff 's translations by cat1 are indeed a major change. For increased flexibility, actually these names (i.e. the original/currently used version of this SC) would have to go into an "almstars user catalog" like "catpick.dat", reprocessed to provide the current Greek/English strings as default solution, and advanced users may then run the name extractor of their choice to get the other versions. (Is the Unicode Greek a problem for the catalog format?) When the "UserStarCatalog" plugin becomes reality, these 5 catalogs would give a first set of examples. (There is also a 1538 (?) Greek print of the Almagest available, which "somebody" could make accessible as yet another version...)

abalkin commented:
> The replacement of @sushoff 's translations by cat1 are indeed a major change.

@gzotti – I restored  @sushoff 's translations for the 16 proper names

![image](https://user-images.githubusercontent.com/535197/118508451-35c4c080-b6fd-11eb-9ca4-4c114e2720fd.png)

While I disagree with some choices, changing existing names is outside of the scope of this PR.  I will open separate issues to discuss possible changes.

sushoff commented:
> While I disagree with some choices, changing existing names is outside of the scope of this PR. I will open separate issues to discuss possible changes.

Thank you! 
I will be happy to discuss the translations: we already discussed this at several occations but not as consequently as I would wish to. It always helps to develop a common sense on how to express something verbally.

abalkin commented:
> Suggestion: Maybe mark the "unformed" stars separately, like "Boo U1"?

@gzotti  - in 5b825af85288ccd54ece87d101a1f0ada624e2ac, I  changed "Con n" labels for the stars outside of the figures to lowercase "con n".   I would like to avoid making these tags longer even by a few characters.  As an alternative, we can use some typographic solutions – for example render these tags in italic, but lowercase is the simplest and probably good enough solution.

In the script, I preserved Toomer's  distinction between stars "near", "under", "around", "round" (?) and simply "outside" constellations.  This information is currently not used, but may become useful in the future.

abalkin commented:
> The newly developed format of "computational books" (cBooks) shall be used for this.

@sushoff - I have not heard of cBooks.  Can you point me to a description of this new format?

sushoff commented:
> @sushoff - I have not heard of cBooks. Can you point me to a description of this new format?

unfortunately not yet: we are currently editing the first books of this format (a conference proceedings that will hopefully appear this year and the Euler edition). cBook format is developed by Gerd Grasshoff and his colleagues at Max Planck Institute for the history of science in Berlin: he is currently introducing it with his newly founded publisher.

The clou is a combination of  
  * a digital book written in LaTeX 
  * with annotations, hyperlinks and all the known nice modern applications (from kindle etc.)
  * plus implemented Jupyter Notebooks for all types of code and visualisations

abalkin commented:
>  (Are the catalogs completely compatible this way?)

@gzotti - I tested the script on the four almstars catalogues and got the following diagnostic:

```
Almagest Catalog:  Toomer/Grasshoff
  found 1022 unique stars and 16 proper names
52 nu1 Boo (HR 5763) is in Boo, Her.
112 bet Tau (HR 1791) is in Aur, Tau.
24 alp PsA (HR 8728) is in Aqr, PsA.
```

```
Almagest Catalog:  Peters/Knobel
  found 1018 unique stars and 16 proper names
52 nu1 Boo (HR 5763) is in Boo, Her.
112 bet Tau (HR 1791) is in Aur, Tau.
24 Cap (HR 8080) is in Cap, PsA.
24 alp PsA (HR 8728) is in Aqr, PsA.
```

```
Almagest Catalog:  Manitius
  found 1011 unique stars and 16 proper names
52 nu1 Boo (HR 5763) is in Boo, Her.
112 bet Tau (HR 1791) is in Aur, Tau.
87 mu Cet (HR 813) is in Ari, Cet.
24 alp PsA (HR 8728) is in Aqr, PsA.
```

```
Almagest Catalog:  Pickering
  found 1021 unique stars and 16 proper names
52 nu1 Boo (HR 5763) is in Boo, Her.
112 bet Tau (HR 1791) is in Aur, Tau.
24 alp PsA (HR 8728) is in Aqr, PsA.
19 phi2 Cet (HR 235) is in Cet, Cet.
17 phi1 Cet (HR 194) is in Cet, Cet.
```

gzotti commented:
Good. At least all cats can be processed. Just make sure to add processing notes in the README.md for the benefit of future developers. It may still be years until that user catalog plugin becomes reality, but I think these lists should really be handled separately from the "user-friendly" skycultures at some later date. (Or else skycultures become unmanageable.) There is always room for improvement.

abalkin commented:
> make sure to add processing notes in the README.md for the benefit of future developers. 

Will do.

> It may still be years until that user catalog plugin becomes reality, but I think these lists should really be handled separately from the "user-friendly" skycultures at some later date.

According to the current [info.ini](https://github.com/Stellarium/stellarium/blob/04b444e3941a1ea7759583e40bc0a69f34783728/skycultures/almagest/info.ini#L5) file, "Almagest" is a single-source SC with Toomer listed as the only source:
 
```
source = Ptolemy's Almagest translated by G. Toomer (1984) plus author's own research on visualization
license = CC BY-ND 4.0 International
classification = single
```

I would like to keep it that way for now and focus on making all information that can be found in Toomer faithfully presented in Stellarium  sky displays.

kajaji commented:
> I would like to keep it that way for now and focus on making all information that can be found in Toomer faithfully presented in Stellarium sky displays.

Please note that Toomer/Graßhoff stars identifications is not the same as in Toomer tables.
This a list of differences I found:

![GrasshofxToomer](https://user-images.githubusercontent.com/34771982/118568088-c217c780-b77f-11eb-967e-6376bbf40d0f.JPG)

kajaji commented:
> > @alex-w IMHO these entries should be translatable for the sake of non-western alphabet languages. For example a casual Arab user will not relate "Her 30" to the corresponding Arabic name "Al-Jathi".
> 
> @kajaji - I think you are looking at this trough the prism of al-Sufi rather than Almagest SC. In al-Sufi SC, translating "Con n" labels to Arabic would make perfect sense because you can use Abjad numerals for the 'n' part which are native to the source. In Almagest SC, these labels are purely navigational and don't have any historical significance.
> 
> What I can do is include the list of abbreviations together with full constellation names in the description. If I do it in say 4-column table, the result will be relatively short (12 rows). This way, once description is translated to users' language, they will have a handy reference to look up.

@abalkin Even in Al-Sufi SC the labels are navigational. To label the stars with Abjad numerals as in Al-Sufi tables will make it hard for the user to decipher the original value. Anyhow, my intention is that to have the information on the screen completely in the target translation language rather than in English for the labels and Arabic elsewhere.

abalkin commented:
> To label the stars with Abjad numerals as in Al-Sufi tables will make it hard for the user to decipher the original value.

@kajaji  - my suggestion of keeping the original abjad labeling was limited to Arabic translation, but on the other hand, in other languages, you can provide transliterations, for example "Alif in UMi" is short enough label that retains the flavor of the source and is distinct from Flamsteed or Bayer notations.

Do I understand correctly that at least in some surviving copies of Al-Sufi's book of fixed stars, smaller stars ate labeled with Abjad numerals in the drawings  while larger have their proper Arabic names written to them?  I would think that for the  best experience of an Arabic-speaking user, the Stellarium sky with Al–Sufi SC and Arabic SC–language selected should look as close as possible to the drawings in the book including star labels.

kajaji commented:
> > To label the stars with Abjad numerals as in Al-Sufi tables will make it hard for the user to decipher the original value.
> 
> @kajaji - my suggestion of keeping the original abjad labeling was limited to Arabic translation, but on the other hand, in other languages, you can provide transliterations, for example "Alif in UMi" is short enough label that retains the flavor of the source and is distinct from Flamsteed or Bayer notations.
> 
> Do I understand correctly that at least in some surviving copies of Al-Sufi's book of fixed stars, smaller stars ate labeled with Abjad numerals in the drawings while larger have their proper Arabic names written to them? I would think that for the best experience of an Arabic-speaking user, the Stellarium sky with Al–Sufi SC and Arabic SC–language selected should look as close as possible to the drawings in the book including star labels.

@abalkin The star numbers in Al-Sufi book are mentioned four times:
- In the general description of the constellation where each star is described by first mentioning its index, e.g 1st or 14th.
- In mapping the genuine Arab star names to the stars in the constellation, like saying "The 14th star of Taurus is called by the Arabs Al-Dabaran"
- In the images using the Abjad system for each star.
- In the tables using the Abjad system for each star.
We can use the Abjad system as used in the manuscripts but I think it is not helpful for the interested user who is reading the general description section  to identify the stars.

In the manuscripts all the stars all labelled with Abjad numerals. Stars that have proper names or Asterisms have the name added along with the Abjad numerals.

abalkin commented:
> Please note that Toomer/Graßhoff stars identifications is not the same as in Toomer tables.

I checked the first three entries in your table and it looks like it is a case of a typo in Toomer.  Looking at the printed table (page 349) I see that Her 20-22 are identified as 74(x), 77(y) and 82(z) Her.  But x Her is actually 77, y Her is 82 and z Her is 88. 

As I am describing in README.md that the actual source is Toomer/Graßhoff.  I don't think the differences are important enough to be mentioned in the description, but polishing the description is beyond the scope of this PR.

alex-w commented:
@sushoff @gzotti please review the PR

abalkin commented:
GitHub tells me that there is a requested change, but I cannot figure out what it is.

alex-w reviewed (commented):
This list of objects was removed from `dso_names.fab` file: M 45, Mel 25, Mel 111, NGC 6441,
Mel 180

Why?

alex-w commented:
> @gzotti - in [5b825af](https://github.com/Stellarium/stellarium/commit/5b825af85288ccd54ece87d101a1f0ada624e2ac), I changed "Con n" labels for the stars outside of the figures to lowercase "con n". I would like to avoid making these tags longer even by a few characters. 

Why not just add an asterisk to the designation? In any way the designation "con n" or "Con n*" should be described in the description.en.utf8 file.

> As an alternative, we can use some typographic solutions – for example render these tags in italic, but lowercase is the simplest and probably good enough solution.

Maybe in the future we may add tool to use some typographic solutions, but right now it's not possible.

abalkin commented:
> Why not just add an asterisk to the designation? In any way the designation "con n" or "Con n*" should be described in the description.en.utf8 file.

@alex-w - I added a paragraph to the description file describing the "Con n" designations.  (See 5d1916599c54bba90003ff33cc1fdfc32f2680d4.) I  still like lowercase more than adding a star because "outside of the figure" stars tend to be fainter and less important.  A notable exception is Arcturus, but it has a proper name, so users will only see the "boo 23" if they click on "Arcturus".  In any case, this is something that can be easily changed in the future after we reach a consensus on how this should be done across Almagest-like sky cultures.

abalkin commented:
> This list of objects was removed from `dso_names.fab` file: M 45, Mel 25, Mel 111, NGC 6441, Mel 180.
> 
> Why?

@alex-w - a short and honest answer is because these entries did not appear in the Write's table.  I can easily add those back, but I am not sure that would be historically accurate.  Let's consider the cases of M 45 (Pleiades) and Mel 25 (Hyades).   Unlike astronomers of telescope era, Ptolemy did not identify Pleiades or Hyades as star clusters.  For him, those were just asterisms.

Copying from Toomer,

```
30-33 The Pleiades:
30                  the northern end of the advance side
31                   the southern end of the advance side
32                  the rearmost and narrowest end of the Pleiades
33.                 the small star outside' the Pleiades, towards the north
```

```
11-15 The stars in the face, called ‘the Hyades’:
11                 the one on the nostrils
12                 the one between this and the northern eye
13                 the one between it [no. 11] and the southern eye
14                 the bright star of the Hyades, the reddish one on the southern eye
15                 the remaining one, on the northern eye
```

So, I don't think it is correct to identify Ptolemy's Pleiades  and Hyades with M 45 and Mel 25 open clusters.  Instead, these should be listed as asterisms.  This said, changing the treatment of these objects is outside of the scope of this PR and I can easily restore the existing associations pending future developments.  Please let me know how you would like to proceed.

abalkin commented:
> This list of objects was removed from dso_names.fab file: [...], Mel 111, NGC 6441, Mel 180.

For completeness, let me address the rest of the deleted DSO names. The last two entries refer to the same cluster and I think in this case we should at most keep the NGC association with the Almagest description.

##  Mel 111 (Coma Star Cluster)

A strong case can be made for keeping this.  In Toomer, we find a mention of Coma in the description of Leo 33
```
33    The northernmost part of the nebulous mass between the edges of Leo
      and Ursa |Major|, called Coma [Berenices]

```

and in the summary of stars in Leo:

"5 stars, 1 of the fourth magnitude, 4 of the fifth, plus Coma"

##  Mel 180, NGC 6441 (globular cluster in Scorpius)

```
22    The nebulous star lo the rear of the sting
```

This PR associates "sco 22" with G Sco, which is correct, but Toomer also says the following in the footnote:

"Manitius identifies this as ('G Scorpii, P-K as gamma Telescopii, an obsolete designation which is the same as G Scorpii (BSC 6630). But the description ‘nebulous’ obviously includes the globular cluster (cf. P -K no. 567 p. 105 and Burnham III 1689)."

So, I am inclined to restore the "NGC 6441" association, but I think "Mel 180" is strictly redundant.

abalkin commented:
I did not hear back from @alex-w, so in eeb1d1cac24f26ac89439a119ecaf6f543038858, I conservatively restored the DSO names including Pleiades and Hyades.

sushoff commented:
> Ptolemy did not identify Pleiades or Hyades as star clusters. For him, those were just asterisms.

well this is only half of the truth! Of course, these clusters were considered as star clusters - but the concept of star clusters was different those days.As stars were not considered as bodies in space with an evolution, they did not consider a group of gas balls in space but a grouping of dots on the celestial globe. 

Indeed this view of the abovementioned clusters (Pleiads, Hyades, Com, Perseus...) had been given since >2000 years at this time. Ptolemy in particular wrote a star catalogue! He needed _points_ in the mathematical sense to address with point-coordinates - the concept of areas on a curved plain was not yet developed, he did not have a mathematics to describe it. 

The concept of star clusters did not fit into a star catalogue. The Pleiades were considered as "seven sisters" but he lists only 4 stars because this way, the area was clearly defined. He does not aim to list all stars that are seen on the sky but he aimed to define the positions of the constellation areas.

Therefore, I don't consider it wrong to delete these clusters. I labeled the clusters as a total because they were already named as a group in antiquity - no matter whether or not Ptolemy listed some single star members in his catalogue. I think, the label "M45" for a modern hobby astronomer designates the same as "Pleiades" for Ptolemy, Hipparchus, Ovid, Hyginus, Aratos ... and all the others.

alex-w commented:
@sushoff is this PR ready for merging by your opinion?

sushoff reviewed (changes requested):
there is an important diacritical sign missing at the beginning of Ὑάδες

alex-w commented:
Thanks for fix @abalkin 

Please reiview the changes @sushoff

sushoff commented:
> Thanks for fix @abalkin
> 
> Please reiview the changes @sushoff

yes, for the moment, it's ok, I think

alex-w commented:
@sushoff OK, thanks! Can I merge it now?

sushoff commented:
please do so
